### PR TITLE
Cleanup more of rustdoc

### PR DESCRIPTION
**Pull request by jyn514**
-  Use `Item::from_def_id` for StructField 
- Use `from_def_id_and_parts` for primitives and keywords
- Take `String` instead of `Symbol` in `from_def_id` - this avoids having to intern then immediately stringify the existing string.
- Remove unused `get_stability` and `get_deprecation`
- Remove unused `attrs` field from `primitives`
- Remove unused `attrs` field from `keywords`

This will probably conflict with https://github.com/rust-lang/rust/pull/79335 and I would prefer for that PR to land first - I'm anxious for https://github.com/rust-lang/rust/pull/77467 to land :)

Makes https://github.com/rust-lang/rust/issues/76998 easier to add.

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustdoc/clean/inline.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ crate fn try_inline(
     let attrs = merge_attrs(cx, Some(parent_module), target_attrs, attrs_clone);
 
     cx.renderinfo.borrow_mut().inlined.insert(did);
-    let what_rustc_thinks = clean::Item::from_def_id_and_parts(did, Some(name), kind, cx);
+    let what_rustc_thinks = clean::Item::from_def_id_and_parts(did, Some(name.clean(cx)), kind, cx);
```
Comment (pickfire) [inline src/librustdoc/clean/inline.rs:127]: I haven't look deep but I wonder why do we need to have this `.clean(cx)` so many times unlike before?

Comment (jyn514) [reply src/librustdoc/clean/inline.rs:127]: Does the explanation in the commit message help? https://github.com/rust-lang/rust/pull/79372/commits/bbded0bf2a59d6d0f53f8f5c2d0b0d944ef38256

Comment (pickfire) [reply src/librustdoc/clean/inline.rs:127]: Yes, now I understand. I didn't know it is related to interning.

**Comment by jyn514** [comment]
This is ready for review.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Awesome as usual, thanks!

@bors: r+

**Pull request merged**